.github/workflows/determine-workflow-runner.yml

```
@@ -0,0 +1,49 @@
+name: Determine Workflow Runner group
+
+on:
+  workflow_call:
+    inputs:
+      default_runner:
+        description: The runner type that is used by the calling workflow by default
+        required: true
+        type: string
+    outputs:
+      runner_group:
+        description: The runner all subsequent jobs within the calling workflow should run on
+        value: ${{ jobs.determine_workflow_runner.outputs.runner_group || inputs.default_runner }}
+
+env:
+  LARGE_RUNNER_GROUP_NAME: pl-4-core-large-runner
+
+jobs:
+  determine_workflow_runner:
+    runs-on: >-
+      ${{
+        (
+          github.event_name == 'pull_request'
+          && contains(github.event.pull_request.labels.*.name, 'urgent')
+        ) && 'pl-4-core-large-runner' || 'ubuntu-latest'
+      }}
+
+    outputs:
+      runner_group: ${{ steps.runner_group.outputs.runner_group }}
+
+    steps:
+      - name: Output Runner Group name
+        if: >-
+          ${{
+            github.event_name == 'pull_request'
+            && startsWith(inputs.default_runner, 'ubuntu')
+          }}
+        id: runner_group
+        env:
+          # We are not able to use \d to check numeric values as bash does not allow them (not POSIX compliant)
+          RC_BRANCH_FORMAT_REGEX: v[0-9]+\.[0-9]+\.[0-9]+_rc[0-9]?
```

```suggestion
          RC_BRANCH_FORMAT_REGEX: v[0-9]+\.[0-9]+\.[0-9]+\-rc[0-9]?
```

We usually use dashes between the version number and "rc0", not underscores